demo/demo.sh

```diff
@@ -14,14 +14,18 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-cd $(dirname ${BASH_SOURCE})
-. ./udemo.sh
-. ../scripts/util.sh
+# Resolve the absolute path to this script directory.
+script_dir="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+
+# Source the helper scripts with absolute paths.
```

Comment:
I don't think we need to re-iterate that we use an absolute path everywhere tbh, if you want you could add that to the first one when script_dir is defined, WDYT?

Reply:
The proposed commitments have been added. Thank you!